docs/Displaying a User Interface/Popup Keyboards/index.md

```diff
@@ -177,10 +185,190 @@ KeyboardListener keyboardListener = new KeyboardListener() {
+![Numeric Keyboard](assets/keyboard_numeric.png)
+!@
+
+@![android, javaSE, javaEE]
+```java
+KeyboardProperties keyboardConfiguration = new KeyboardProperties()
+        .setKeyboardLayout(KeyboardLayout.NUMERIC);
+
+sdlManager.getScreenManager().setKeyboardConfiguration(keyboardConfiguration);
+```
+!@
+
+@![iOS]
+todo add iOS implementation 
+!@
+
+@![iOS, android, javaSE, javaEE]
+#### Input Masking (RPC 7.1+)
+You can modify the keyboard to mask the entered characters by changing the keyboard configuration's `maskInputCharacters`. 
+![Numeric Keyboard](assets/keyboard_numeric_masked.png)
+!@
+
+@![android, javaSE, javaEE]
+```java
+KeyboardProperties keyboardConfiguration = new KeyboardProperties()
+        .setKeyboardLayout(KeyboardLayout.NUMERIC)
+        .setMaskInputCharacters(KeyboardInputMask.ENABLE_INPUT_KEY_MASK);
+
+sdlManager.getScreenManager().setKeyboardConfiguration(keyboardConfiguration);
+```
+!@
+
+@![iOS]
+todo add iOS implementation 
+!@
+
+@![iOS, android, javaSE, javaEE]
+#### Custom Keys (RPC 7.1+)
+Each keyboard layout has a number of keys that can be customized to your app's needs. For example, you could set two of the customizable keys in `QWERTY` layout to be "!" and "?" as seen in the image below. The available number and location of these custom keys is determined by the connected head unit.
+
+![Custom Keys](assets/keyboard_querty_custom_keys.png)
+!@
+
+@![android, javaSE, javaEE]
+```java
+KeyboardProperties keyboardConfiguration = new KeyboardProperties()
+        .setKeyboardLayout(KeyboardLayout.QWERTY)
+        .setCustomKeys(Arrays.asList("!", "?"));
+
+sdlManager.getScreenManager().setKeyboardConfiguration(keyboardConfiguration);
+```
+!@
+
+@![iOS]
+todo add iOS implementation 
+!@
+
+#### Single-Use Keyboard Configuration
+@![android, javaSE, javaEE]
+If you want to change the keyboard configuration for only one keyboard session and keep the default global keyboard properties unchanged, you can instead pass the `KeyboardProperties` to `presentKeyboard()` which was discussed in the `Presenting a Keyboard` section.
+!@
+
+@![iOS]
+If you want to change the keyboard properties for only one keyboard session and keep the default global keyboard properties unchanged, you can instead return a `SDLKeyboardProperties` in the `SDLKeyboardDelegate.customKeyboardConfiguration` method.
+!@
+
+@![iOS, android, javaSE, javaEE]
+### Checking Keyboard Capabilities (RPC v7.1+)
+Each head unit can support different set of keyboard layouts and each layout can support different number of custom keys. Also, some head unit may not support masking input. If you want to know which keyboard features are supported on the connected head unit, you can check the `KeyboardCapabilities`:
+!@
+@![android, javaSE, javaEE]
+```java
+sdlManager.getSystemCapabilityManager().getCapability(SystemCapabilityType.DISPLAYS, new OnSystemCapabilityListener() {
```

Comment:
I believe you were going to change this to just pull the `SystemCapabilityManager` default main window capability keyboard capabilities.